packages/babel-plugin-transform-modules-umd/src/index.js

```diff
@@ -170,14 +177,25 @@ export default declare((api, options) => {
           }
 
           for (const [source, metadata] of meta.source) {
-            amdArgs.push(t.stringLiteral(source));
+            const {
+              filename: sourceFilename,
+              filenameRelative: sourceRelative,
+            } = resolveRelativeImportPaths(source, fileOpts);
+            amdArgs.push(
+              t.stringLiteral(amdImportId(source, fileOpts, options)),
```

Comment:
For consistency, I think the transform of amd import source should also be applied to https://github.com/babel/babel/blob/866a742bf79c01513d2ffd7ef55ca6be3fd9c650/packages/babel-plugin-transform-modules-amd/src/index.js#L121

Reply:
Ah right, I was struggling to understand for a moment but then I saw it was about the `amd` and `commonjs` module plugins... First I wanted to see how the reaction would be, but yeah ofc we should make them behave the same way if these changes are accepted 🙂